code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,409 @@
+// ***************************************
+// *********** Psychic shield
+// ***************************************
+/datum/action/xeno_action/activable/psychic_shield
+	name = "Psychic Shield"
+	ability_name = "psychic shield"
+	action_icon_state = "psy_shield"
+	mechanics_text = "Creates a protective field of psychic energy in front of you."
+	cooldown_timer = 8 SECONDS
+	plasma_cost = 200
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PSYCHIC_SHIELD,
+	)
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
+	use_state_flags = XACT_USE_BUSY
+	var/obj/effect/xeno/shield/active_shield
+
+
+/datum/action/xeno_action/activable/psychic_shield/on_cooldown_finish()
+	to_chat(owner, span_notice("We gather enough mental strength to create a psychic shield again."))
+	return ..()
+
+/datum/action/xeno_action/activable/psychic_shield/use_ability(atom/A)
+	. = ..()
+	if(active_shield)
+		var/mob/living/carbon/xenomorph/xeno_owner = owner
+		if(plasma_cost > xeno_owner.plasma_stored)
+			to_chat(xeno_owner, span_warning("We need [plasma_cost - xeno_owner.plasma_stored] more plasma!"))
+			return FALSE
+		if(can_use_action(FALSE, XACT_USE_BUSY))
+			shield_blast()
+			cancel_shield()
+		return
+	owner.visible_message("A strange and violent psychic aura is suddenly emitted from \the [owner]!")
```

Comment:
```suggestion
	X.visible_message(span_xenodanger("[X] emits strange and violent psychic aura!")!"), \
```
followed by:
`span_xenodanger("Our great mind conjures a reflective shield. We must keep still!"))
X.balloon_alert(X, "Keep still...")`

and before that:
`var/mob/living/carbon/xenomorph/warlock/X = owner`

Reply:
See the defiler ability code.

Reply:
you don't actually need to typecast owner for that